File: libs/python/sorting.py
def insertion_sort(array):
	# starting from the second element
	for i in range(1,len(array)):
		# keep the value that will be  inserted because it will lost
		# when shifty element to the right
		key = array[i] 
		# and start comparing from from the element before it
		j = i - 1 
		# loop to the correct position of key
		while j >= 0 and array[j] > key:
			# shift all elements to the right
			array[j+1]=array[j]
			j -= 1
		# insert key at the correct position in front of the 
		# first element that is not larger than key 
		array[j+1] = key
def insertion_sort_test():
	sorted_data = [int(x) for x in open("./data/sorted_data","r").read().split("\n") if x != '']
	unsorted_data = [int(x) for x in open("./data/unsorted_data","r").read().split("\n") if x != '']
	insertion_sort(unsorted_data)
	assert sorted_data == unsorted_data

File: libs/julia/sorting.jl
function insertion_sort(A::Array{Integer,1})
    for j = 1:length(A)
        key = A[j]
        i = j - 1
        while i > 0 && A[i] > key
            A[i+1] = A[i]
            i = i - 1
        end
        A[i+1]=key
    end
end
function insertion_sort_test()
    sorted_data = [parse(Int,x) for x in split(readall("data/sorted_data"),['\n'])]
    unsorted_data = [parse(Int,x) for x in split(readall("data/unsorted_data"),['\n'])]
    insertion_sort(unsorted_data)
    if sorted_data != unsorted_data
        println("didn't pass")
    else
        println("passed")
    end
end
insertion_sort_test()